components/info/impl/src/main/java/com/flipperdevices/info/impl/viewmodel/FullInfoViewModel.kt

```
@@ -0,0 +1,135 @@
+            }.launchIn(viewModelScope)
+
+        serviceApi.flipperRpcInformationApi.getRpcInformationFlow().onEach { rpcInformation ->
+            deviceInfoState.update {
+                it.copy(
+                    flashInt = rpcInformation.internalStorageStats,
+                    flashSd = rpcInformation.externalStorageStats
+                )
+            }
+            flipperRpcInformationState.emit(rpcInformation)
+        }.launchIn(viewModelScope)
+
+        serviceApi.flipperRpcInformationApi.getRequestRpcInformationStatus().onEach {
+            info { "FlipperRequestRpcInformationStatus: $it" }
+            when (it) {
+                is FlipperRequestRpcInformationStatus.InProgress ->
+                    deviceInfoRequestStatus.emit(DeviceInfoRequestStatus(it))
+                FlipperRequestRpcInformationStatus.NotStarted ->
+                    deviceInfoRequestStatus.emit(DeviceInfoRequestStatus())
+            }
+        }.launchIn(viewModelScope)
+    }
+
+    fun getFirmwareChannel(commit: String?): FirmwareChannel? {
+        if (commit == null) return null
+        val preparedCommit = commit.split(" ")
+        if (preparedCommit.isEmpty()) return null
+        val branch = preparedCommit.first()
+        return firmwareVersionBuilderApi.getFirmwareChannel(branch)
+    }
+
+    fun shareDeviceInfo() {
+        val context = getApplication<Application>().applicationContext
+
+        val nameFile = getFileName()
+        val file = File(FlipperStorageProvider.getSharedKeyFolder(context), nameFile)
+        file.createClearNewFileWithMkDirs()
+
+        try {
+            addInfoToFile(file, context)
+            ShareHelper.shareFile(
+                context = context,
+                file = file,
+                name = nameFile,
+                resId = R.string.device_info_share
+            )
+        } catch (@Suppress("SwallowedException") exception: Exception) {
+            error { "Exception when upload device info" }
```

You shouldn't suppress exception, print it